pkg/sources/source_manager.go

```diff
@@ -312,17 +311,18 @@ func (s *SourceManager) runWithUnits(ctx context.Context, handle handle, source
 		unitPool.SetLimit(s.concurrentUnits)
 	}
 	for unit := range reporter.unitCh {
+		unit := unit
 		reporter := &mgrChunkReporter{
-			unitID:  unit.SourceUnitID(),
+			unit:    unit,
```

Comment:
why did you need to switch this to be a full unit?

Reply:
The `report` object has a method `AddChunkError` that needs a full unit. I think it logically makes sense even though we'll only use the `SourceUnitID() string`